account-kit/signer/src/base.ts

```diff
@@ -170,11 +201,16 @@ export abstract class BaseAlchemySigner<TClient extends BaseSignerClient>
    * @returns {Promise<User>} the user that was authenticated
    */
   authenticate: (params: AuthParams) => Promise<User> = async (params) => {
-    if (params.type === "email") {
-      return this.authenticateWithEmail(params);
+    switch (params.type) {
+      case "email":
+        return this.authenticateWithEmail(params);
+      case "passkey":
+        return this.authenticateWithPasskey(params);
+      case "oauth":
+        return this.authenticateWithOauth(params);
+      case "oauthReturn":
+        return this.handleOauthReturn(params);
```

Comment:
nit: should we add a `default` which throws? Actually maybe not because I remember something similar in Rust where we only handle the cases that exist so that the compiler throws an error when something isn't handled right?

Reply:
Yea, when it's written like this, if a new option is added for `params.type` or if we forgot to handle one of them, then this will become a compiler error until we add handling for the new type.

Something I've seen done before that we might consider, which maintains this typechecking but also throws a descriptive runtime error, is to define an `assertNever` utility function. It would look something like
```ts
function assertNever(x: never, makeMessage: (x: unknown) => string): never {
  throw new Error(makeMessage(x as unknown));
}
```
and used
```ts
switch (params.type) {
  case "email":
    // ...
  default:
    assertNever(params.type, (x) => `Unknown auth type: ${x}`);
}
```
You don't need that `makeMessage` as a callback, you could have it just take a string, but then it's annoying because the typesystem doesn't like you interpolating a `never` value into a string so you have to cast it to `any` or `unknown`.

Reply:
Added `assertNever` in a bunch of places, should be nice.

Reply:
nice, yea I like this a lot